src/main/java/domain/Persons.java

```diff
@@ -1,45 +1,64 @@
 package domain;
 
 import exception.ErrorCode;
-import java.util.Collections;
 import java.util.List;
+import java.util.Set;
 import java.util.stream.Collectors;
 
 public class Persons {
     private final List<Person> persons;
 
-    public Persons(List<Person> persons) {
-        validate(persons);
+    private Persons(List<Person> persons) {
         this.persons = persons;
     }
 
-    private void validate(List<Person> persons) {
-        if (isNameDuplicate(persons)) {
+    public static Persons from(List<String> names) {
+        validate(names);
+        return initializePerson(names);
+    }
+
+    private static void validate(List<String> names) {
+        if (isNameDuplicate(names)) {
             throw new IllegalArgumentException(ErrorCode.NAME_DUPLICATE.getMessage());
         }
     }
 
-    private boolean isNameDuplicate(List<Person> names) {
-        return names.size() != names.stream()
-                .map(Person::getName)
-                .distinct()
-                .count();
+    private static boolean isNameDuplicate(List<String> names) {
+        return names.size() != Set.copyOf(names).size();
+    }
+
+    private static Persons initializePerson(List<String> names) {
+        List<Person> persons = names.stream()
+                .map(name -> Person.of(name, names.indexOf(name)))
+                .collect(Collectors.toList());
+        return new Persons(persons);
+    }
+
+    public void playGame(Ladder ladder) {
+        for (Person person : persons) {
+            person.climb(ladder);
+        }
     }
 
-    public List<String> getAllPersonNames() {
+    public int getTotalPersonCount() {
+        return persons.size();
+    }
+
+    public List<String> getAllPersonName() {
         return persons.stream()
                 .map(Person::getName)
                 .collect(Collectors.toList());
     }
 
-    public int getLongestPersonNameLength() {
-        List<Integer> namesLength = getAllPersonNames().stream()
-                .map(String::length)
+    public List<Integer> getAllPersonPosition() {
+        return persons.stream()
+                .map(Person::getPosition)
                 .collect(Collectors.toList());
-        return Collections.max(namesLength);
     }
 
-    public int getTotalPersonCount() {
-        return persons.size();
+    public String findPersonNameInPosition(int position) {
+        return persons.stream()
+                .filter(person -> person.getPosition() == position)
+                .findAny().get().getName();
```

Comment:
```suggestion
        return persons.stream()
            .filter(person -> person.getPosition() == position)
            .map(Person::getName)
            .findAny()
            .orElse(null);
```
Optional로 반환되는경우 get()을 사용하기보다는 orElse(null) 혹은 orElseThrow() 사용해주는게 좋을것 같아요